### PR TITLE
cpu_model_negative: Add new test case

### PR DESCRIPTION
**Pull request by vivianQizhu**
id: 1889249, 1889259, 1889261
Signed-off-by: Qianqian Zhu <qizhu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
Depends on https://github.com/avocado-framework/avocado-vt/pull/2787

**Comment by vivianQizhu** [comment]
(1/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: PASS (4.09 s)
 (2/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.q35: PASS (4.12 s)
 (3/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: FAIL: Qemu should fail to boot but it succeeds. (30.32 s)

@nanliu-r The last one failed, smp=241 boot up successfully, would you help check if the case steps is correct?

**Comment by vivianQizhu** [comment]
@nanliu-r Seems because I test with q35 and different machine type have different value for maxcpus, it's updated and please help review, thanks.

File: qemu/tests/cfg/cpu_model_negative.cfg [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            vcpu_maxcpus = 4
+            warning_msg = 'maxcpus must be equal to or greater than smp'
+        - smp_out_of_range:
+            enforce_smp_topology = yes
+            smp = OUT_OF_RANGE
+            vcpu_cores = 0
+            vcpu_threads = 0
+            vcpu_sockets = 0
+            vcpu_maxcpus = 0
+            warning_msg = 'Number of SMP cpus requested (${smp}) exceeds max cpus supported by KVM'
```
Comment (nanliu-r) [inline qemu/tests/cfg/cpu_model_negative.cfg:24]: @vivianQizhu Hi, This is the qemu warning  info: 
[qemu output] qemu-kvm: Invalid SMP CPUs 513. The max CPUs supported by machine 'pc-q35-rhel8.3.0' is 512.

Comment (vivianQizhu) [reply qemu/tests/cfg/cpu_model_negative.cfg]: @nanliu-r Updated, thanks.
 (1/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: PASS (4.08 s)
 (2/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.q35: PASS (4.14 s)
 (3/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: PASS (4.16 s)

**Comment by vivianQizhu** [comment]
@nanliu-r Updated, please help review, thanks.
 (1/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.enforce.q35: PASS (2.18 s)
 (2/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_lt_maxcpu.q35: PASS (2.10 s)
 (3/3) Host_RHEL.m8.u3.product_rhel.ovmf.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.cpu_model_negative.smp_out_of_range.q35: PASS (2.14 s)

File: qemu/tests/cpu_model_negative.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Runs CPU negative test:
+
+    1. Boot guest with improper cpu configuration
+    2. Verify guest boot fail
```
Comment (nanliu-r) [inline qemu/tests/cpu_model_negative.py:18]: Should be 'Boot qemu', and 'qemu boot fail'.

File: qemu/tests/cpu_model_negative.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        m_type = re.search(r'\(alias of (\S+)\)', m_types)[1]
+        max_value = utils_qemu.get_maxcpus_hard_limit(qemu_bin, m_type)
+        smp = str(max_value + 1)
+        params['wrong_cmd'] = params['wrong_cmd'].replace(
+            'MACHINE_TYPE', machine_type).replace('OUT_OF_RANGE', smp)
+        msg = params['warning_msg'].replace('SMP_VALUE', smp).replace(
+            'MAX_VALUE', str(max_value)).replace('MACHINE_TYPE', m_type)
+        params['warning_msg'] = msg
+    warning_msg = params['warning_msg']
+    wrong_cmd = '%s %s' % (qemu_bin, params['wrong_cmd'])
```
Comment (nanliu-r) [inline qemu/tests/cpu_model_negative.py:44]: Can we add one log to print what's the `wrong_cmd` executed?

File: qemu/tests/cpu_model_negative.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    status, output = process.getstatusoutput(wrong_cmd)
```
Comment (nanliu-r) [inline qemu/tests/cpu_model_negative.py:45]: Can we add one log to print what's the `output` in case we get one warning which is not expected?

**Comment by vivianQizhu** [comment]
@nanliu-r All updated, thanks.

File: qemu/tests/cfg/cpu_model_negative.cfg [hunk 5]
```diff
@@ -0,0 +1,14 @@
+- cpu_model_negative:
+    type = cpu_model_negative
+    start_vm = no
+    warning_msg = ''
+    variants:
+        - enforce:
+            wrong_cmd = "-cpu 'Skylake-Client',+avx512_bf16,check,enforce"
```
Comment (nanliu-r) [inline qemu/tests/cfg/cpu_model_negative.cfg:7]: we need to find one available cpu model. Others look good to me.

Comment (vivianQizhu) [reply qemu/tests/cfg/cpu_model_negative.cfg]: @nanliu-r Updated, thanks.

File: qemu/tests/cpu_model_negative.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    1. Launch qemu with improper cpu configuration
+    2. Verify qemu failed to start
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    if ('CPU_MODEL' in params['wrong_cmd'] and
+            '+avx512_bf16' in params['cpu_model_flags']):
```
Comment (nanliu-r) [inline qemu/tests/cpu_model_negative.py:26]: seems we won't get the flag `avx512_bf16` in `params['cpu_model_flags']`, it is not defined in cfg.

Comment (vivianQizhu) [reply qemu/tests/cpu_model_negative.py]: @nanliu-r Updated, thanks.

**Comment by nanliu-r** [comment]
Thanks.
ACK.

**Pull request merged**